controlplane/kubeadm/internal/controllers/remediation.go

```diff
@@ -368,7 +368,7 @@ func (r *KubeadmControlPlaneReconciler) reconcileUnhealthyMachines(ctx context.C
 		controlplanev1.RemediationInProgressAnnotation: remediationInProgressValue,
 	})
 
-	return ctrl.Result{Requeue: true}, nil
+	return ctrl.Result{}, nil // No need to requeue here. Machine deletion above triggers reconciliation.
```

Comment:
```suggestion
	return ctrl.Result{RequeueAfter: time.Millisecond}, nil // Technically there is no need to requeue here. Machine deletion above triggers reconciliation. But we have to return a non-zero Result so reconcile above returns.
```
My bad. We have to requeue here otherwise the reconcile func one level higher does not return which it absolutely has to

(using 1 ms so we don't get an additional requeue on top of the requeue we get from Machine deletion. With e.g. 10s we might get an additional requeue)

Reply:
I also missed it. Thanks!